меню

frozzzen123 opened the pull request:
Создал начальную визуализацию меню в состоянии по default. Т.е. создано отображение меню без какого-либо взаимодействия с пользователем. В конце разработки проекта картинки будут переработаны (улучшен дизайн). Предложения по стилизации приветствуются

frozzzen123 committed:
Update Python.cpp

frozzzen123 committed:
Update Python.h

frozzzen123 committed:
added pictures for menu

23350406 commented on Python/src/Python.h:
```diff
@@ -140,7 +146,7 @@ class Layout
 		// Из файла считывается шрифт Consolas.
 		sf::Font consolas;
-		consolas.loadFromFile("../../../fonts/Consolas.ttf");
+		consolas.loadFromFile("C:/Users/Admin/source/repos/SnakeGame/fonts/Consolas.ttf");
```
относительные пути

23350406 commented on Python/src/Python.cpp:
```diff
@@ -18,12 +23,11 @@ int main()
 	// Из файла считывается шрифт Consolas.
 	sf::Font consolas;
-	consolas.loadFromFile("../../../fonts/Consolas.ttf");
+	consolas.loadFromFile("C:/Users/Admin/source/repos/SnakeGame/fonts/Consolas.ttf");
```
нужен относительный путь

23350406 commented on Python/src/Python.cpp:
```diff
@@ -1,15 +1,20 @@
 	  К примеру маска цветов - можем игнорировать какие либо цвета объекта*/
 	sf::Image pythonImage;
-	pythonImage.loadFromFile("../../../images/pngwing.com.png");
+	pythonImage.loadFromFile("C:/Users/Admin/source/repos/SnakeGame/images/pngwing.com.png");
```
путь

23350406 commented on Python/src/Python.h:
```diff
@@ -221,3 +227,58 @@ class Layout
+// Функция выводит кнопку "авторы" на экран.
+void DrawAuthors(sf::RenderWindow& window) {
+	sf::Image buttonImg;
+	buttonImg.loadFromFile("C:/Users/Admin/source/repos/SnakeGame/images/authors.png");
```
путь

23350406 commented on Python/src/Python.h:
```diff
@@ -221,3 +227,58 @@ class Layout
+// Функция выводит кнопку "выйти" на экран.
+void DrawLeave(sf::RenderWindow& window) {
+	sf::Image buttonImg;
+	buttonImg.loadFromFile("C:/Users/Admin/source/repos/SnakeGame/images/leave.png");
```
путь

23350406 commented on Python/src/Python.h:
```diff
@@ -221,3 +227,58 @@ class Layout
+// Функция выводит кнопку "настройки" на экран.
+void DrawSettings(sf::RenderWindow& window) {
+	sf::Image buttonImg;
+	buttonImg.loadFromFile("C:/Users/Admin/source/repos/SnakeGame/images/settings.png");
```
путь

23350406 commented on Python/src/Python.h:
```diff
@@ -221,3 +227,58 @@ class Layout
+// Функция выводит кнопку "старт" на экран.
+void DrawStart(sf::RenderWindow& window) {
+	sf::Image buttonImg;
+	buttonImg.loadFromFile("C:/Users/Admin/source/repos/SnakeGame/images/start.png");
```
путь

frozzzen123 committed:
Updated Python.cpp

1. Функции прорисовки главного меню объединены в одну.
2. Добавлена обработка переходов из основного меню в меню начала игры и обратно.
3. Поменял пути на относительные

frozzzen123 committed:
Update Python.h

1. Добавлена функция, объединяющая функции, прорисовывающие главное меню.
2. Добавлена функция прорисовку меню начала игры.
3. Пути изменены на относительные.
4. В создании спрайтов картинок удален процесс создания картинок как объектов sf::Image.

frozzzen123 committed:
Added images

Добавлены картинки для окна меню старта игры.

23350406 commented:
Всё отлично, прогеры работают жёстко

frozzzen123 commented:
Реализовал предварительное меню начала игры
